semantic_router/encoders/vit.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any, List, Optional
+
+from PIL import Image
+from PIL.Image import Image as _Image
+from pydantic.v1 import PrivateAttr
+
+from semantic_router.encoders import BaseEncoder
+
+
+class VitEncoder(BaseEncoder):
+    name: str = "google/vit-base-patch16-224"
+    type: str = "huggingface"
+    score_threshold: float = 0.5
+    processor_kwargs: dict = {}
+    model_kwargs: dict = {}
+    device: Optional[str] = None
+    _processor: Any = PrivateAttr()
+    _model: Any = PrivateAttr()
+    _torch: Any = PrivateAttr()
+    _T: Any = PrivateAttr()
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self._processor, self._model = self._initialize_hf_model()
+
+    def _initialize_hf_model(self):
+        try:
+            from transformers import ViTImageProcessor, ViTModel
+        except ImportError:
+            raise ImportError(
+                "Please install transformers to use HuggingFaceEncoder. "
+                "You can install it with: "
+                "`pip install semantic-router[local]`"
+            )
+
+        try:
+            import torch
+            import torchvision.transforms as T
+        except ImportError:
+            raise ImportError(
+                "Please install Pytorch to use HuggingFaceEncoder. "
+                "You can install it with: "
+                "`pip install semantic-router[local]`"
+            )
+
+        self._torch = torch
+        self._T = T
+
+        processor = ViTImageProcessor.from_pretrained(
+            self.name, **self.processor_kwargs
+        )
```

Comment:
The `ViTImageProcessor` abstracts away the image preprocessing to bring any image to the 224x224 shape needed for ViT